src/index.js

```diff
@@ -18,7 +18,28 @@ const DEFAULT_ALLOW_HEADERS = [
 
 const DEFAULT_MAX_AGE_SECONDS = 60 * 60 * 24 // 24 hours
 
-const cors = (options = {}) => handler => (req, res, ...restArgs) => {
+function isString (s) {
+  return typeof s === 'string' || s instanceof String
+}
+
+function isOriginAllowed (origin, allowedOrigin) {
+  if (Array.isArray(allowedOrigin)) {
+    for (var i = 0; i < allowedOrigin.length; ++i) {
+      if (isOriginAllowed(origin, allowedOrigin[i])) {
+        return true
+      }
+    }
+    return false
+  } else if (isString(allowedOrigin)) {
+    return origin === allowedOrigin
+  } else if (allowedOrigin instanceof RegExp) {
+    return allowedOrigin.test(origin)
+  } else {
+    return !!allowedOrigin
+  }
+}
+
+const cors = (options = {}) => handler => async (req, res, ...restArgs) => {
```

Comment:
I don't think `async` is necessary here, let me know if there's a reason for it I'm not seeing.

Reply:
yes, you are right. must be a leftover from a different approach.

```suggestion
const cors = (options = {}) => handler => (req, res, ...restArgs) => {
```

Reply:
Can you give me permission to push to this repo? Then I can commit this suggested change.

Reply:
I committed it myself now, but gave you access to the repo anyway.